### PR TITLE
rsgislib: update to 5.0.14

### PR DESCRIPTION
**Pull request by nilason**
#### Description

In addition to version bump:
* new Python subports are added (eg. 'py311-rsgislib')
* new port 'rsgislib_select' added to enable default selection of Python executables

Closes: https://trac.macports.org/ticket/67796

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
ping @ryandesign , @reneeotten , @mascguy 

Please review and/or merge. Thanks!

File: gis/rsgislib/Portfile [hunk 1]
```diff
@@ -1,110 +1,164 @@
...
-    configure.args-append   -DPYTHON_EXE=${prefix}/bin/python2.7
-    configure.args-append   -DPYTHON_INSTALL_PREFIX=${destroot}${frameworks_dir}/Python.framework/Versions/2.7/
+# Figure out HDF5's mpi include directory
+set mpl_include_dir ""
+if {![catch {set result [active_variants hdf5 openmpi]}]} {
+    if {$result} {
+        set mpl_include_dir "-I${prefix}/include/openmpi-mp"
+    }
+}
+if {![catch {set result [active_variants hdf5 openmpi-devel]}]} {
```
Comment (mascguy) [inline gis/rsgislib/Portfile:60]: We removed `-devel` subports for `openmpi` and `mpich`, back 2-ish years ago.

File: gis/rsgislib/Portfile [hunk 2]
```diff
@@ -1,110 +1,164 @@
...
+    if {$result} {
+        if {![variant_isset openmpi-devel]} {
```
Comment (mascguy) [inline gis/rsgislib/Portfile:62]: Same comment

File: gis/rsgislib/Portfile [hunk 3]
```diff
@@ -1,110 +1,164 @@
...
+    }
+}
+if {![catch {set result [active_variants hdf5 mpi]}]} {
+    if {$result} {
+        if {![variant_isset mpi]} {
+            set mpl_include_dir "-I${prefix}/include/mpich-mp"
+        }
+    }
+}
+if {![catch {set result [active_variants hdf5 mpich-devel]}]} {
```
Comment (mascguy) [inline gis/rsgislib/Portfile:74]: Same comment

File: gis/rsgislib/Portfile [hunk 4]
```diff
@@ -1,110 +1,164 @@
...
+    if {$result} {
+        if {![variant_isset mpich-devel]} {
```
Comment (mascguy) [inline gis/rsgislib/Portfile:76]: Same comment

**Review by mascguy — CHANGES_REQUESTED** [review]
We removed `-devel` subports for `openmpi` and `mpich`, back 2-ish years ago.

File: gis/rsgislib/files/py311-rsgislib [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisfilelut.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisimg2kmz.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisimg2webtiles.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisimginfo.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisproj.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisshepsegment.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgistranslate2cog.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgistranslate2tif.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisuserpassfile.py
+${frameworks_dir}/Python.framework/Versions/3.11/bin/rsgisvectools.py
```
Comment (mascguy) [inline gis/rsgislib/files/py311-rsgislib:18]: For Python, is anything different about each of these files, apart from the version?

If not, just use one file, with a placeholder for the version. (Something like `@@PYTHON_VER@@`.) Then use `reinplace` to fill that in, as appropriate.

**Review by mascguy — CHANGES_REQUESTED** [review]

**Comment by nilason** [comment]
> We removed `-devel` subports for `openmpi` and `mpich`, back 2-ish years ago.

Ahh, nicked that piece of code... removed the -devel ports.

File: gis/rsgislib/Portfile [hunk 6]
```diff
@@ -1,110 +1,152 @@
...
-    configure.args-append   -DPYTHON_EXE=${prefix}/bin/python2.7
-    configure.args-append   -DPYTHON_INSTALL_PREFIX=${destroot}${frameworks_dir}/Python.framework/Versions/2.7/
+# Figure out HDF5's mpi include directory
+set mpl_include_dir ""
+if {![catch {set result [active_variants hdf5 openmpi]}]} {
+    if {$result} {
+        set mpl_include_dir "-I${prefix}/include/openmpi-mp"
+    }
+}
+if {![catch {set result [active_variants hdf5 mpi]}]} {
```
Comment (mascguy) [inline gis/rsgislib/Portfile:60]: Are you checking whether `hdf5` is built with `mpich`?

If so, the correct variant name is `mpich`.

**Review by mascguy — CHANGES_REQUESTED** [review]

File: gis/rsgislib/Portfile (hunk 6 above)
Comment (nilason) [reply gis/rsgislib/Portfile]: > Are you checking whether `hdf5` is built with `mpich`?

Yes, indeed.

> If so, the correct variant name is `mpich`.

Pushed correction.

(`gdal` Portfile needs a re-work!)

File: gis/rsgislib/Portfile [hunk 7]
```diff
@@ -1,110 +1,152 @@
...
+}
+if {![catch {set result [active_variants hdf5 mpich]}]} {
+    if {$result} {
+        set mpl_include_dir "-I${prefix}/include/mpich-mp"
+    }
+}
+configure.args-append   -DHDF5_INCLUDE_DIR=${prefix}/include
+configure.args-append   -DHDF5_LIB_PATH=${prefix}/lib
+configure.cxxflags-append \
+                        ${mpl_include_dir}
```
Comment (mascguy) [inline gis/rsgislib/Portfile:68]: This is a bit nitpicky, so you needn't change this. (Unless you really, really want to.) But ideally, you would only append `${mpl_include_dir}` if the string isn't empty.

That said, AFAIK, appending an empty string shouldn't cause any issues. So no concerns.

**Comment by mascguy** [comment]
Currently testing on some older macOS releases, to see whether we need to expand the Clang blacklist. More results to follow shortly.

**Review by mascguy — APPROVED** [review]
Tested on macOS 10.12 and 10.15 - to complement the CI coverage of Big Sur and later - and LGTM.

**Comment by nilason** [comment]
@mascguy Thanks a lot!

**Pull request merged**